Shell/main.cpp

```diff
@@ -267,36 +270,53 @@ static int run_command(const String& cmd)
     for (int i = 0; i < subcommands.size(); ++i) {
         auto& subcommand = subcommands[i];
         for (auto& redirection : subcommand.redirections) {
-            if (redirection.type == Redirection::Pipe) {
-                int pipefd[2];
-                int rc = pipe(pipefd);
-                if (rc < 0) {
-                    perror("pipe");
-                    return 1;
+            switch (redirection.type) {
+                case Redirection::Pipe: {
+                    int pipefd[2];
+                    int rc = pipe(pipefd);
+                    if (rc < 0) {
+                        perror("pipe");
+                        return 1;
+                    }
+                    subcommand.redirections.append({ Redirection::Rewire, STDOUT_FILENO, pipefd[1] });
+                    auto& next_command = subcommands[i + 1];
+                    next_command.redirections.append({ Redirection::Rewire, STDIN_FILENO, pipefd[0] });
+                    fds.add(pipefd[0]);
+                    fds.add(pipefd[1]);
+                    break;
                 }
-                subcommand.redirections.append({ Redirection::Rewire, STDOUT_FILENO, pipefd[1] });
-                auto& next_command = subcommands[i + 1];
-                next_command.redirections.append({ Redirection::Rewire, STDIN_FILENO, pipefd[0] });
-                fds.add(pipefd[0]);
-                fds.add(pipefd[1]);
-            }
-            if (redirection.type == Redirection::FileWrite) {
-                int fd = open(redirection.path.characters(), O_WRONLY | O_CREAT, 0666);
-                if (fd < 0) {
-                    perror("open");
-                    return 1;
+                case Redirection::FileWriteAppend: {
```

Comment:
Not super fond of duplication, but I think this is short enough that it's the lesser sin than factoring it out to a function.